clang/utils/TableGen/ClangOpenCLBuiltinEmitter.cpp

```diff
@@ -448,32 +473,32 @@ void BuiltinNameEmitter::EmitExtensionTable() {
 }
 
 void BuiltinNameEmitter::EmitTypeTable() {
-  OS << "static const OpenCLTypeStruct TypeTable[] = {\n";
+  OS << "const " << ClassName << "::ProgModelTypeStruct " << ClassName
+     << "::TypeTable[] = {\n";
   for (const auto &T : TypeMap) {
     const char *AccessQual =
         StringSwitch<const char *>(T.first->getValueAsString("AccessQualifier"))
-            .Case("RO", "OCLAQ_ReadOnly")
-            .Case("WO", "OCLAQ_WriteOnly")
-            .Case("RW", "OCLAQ_ReadWrite")
-            .Default("OCLAQ_None");
+            .Case("RO", "AQ_ReadOnly")
+            .Case("WO", "AQ_WriteOnly")
+            .Case("RW", "AQ_ReadWrite")
+            .Default("AQ_None");
 
     OS << "  // " << T.second << "\n"
-       << "  {OCLT_" << T.first->getValueAsString("Name") << ", "
+       << "  {TID_" << T.first->getValueAsString("Name") << ", "
        << T.first->getValueAsInt("VecWidth") << ", "
        << T.first->getValueAsBit("IsPointer") << ", "
        << T.first->getValueAsBit("IsConst") << ", "
-       << T.first->getValueAsBit("IsVolatile") << ", "
-       << AccessQual << ", "
+       << T.first->getValueAsBit("IsVolatile") << ", " << AccessQual << ", "
        << T.first->getValueAsString("AddrSpace") << "},\n";
   }
   OS << "};\n\n";
 }
 
 void BuiltinNameEmitter::EmitSignatureTable() {
   // Store a type (e.g. int, float, int2, ...). The type is stored as an index
-  // of a struct OpenCLType table. Multiple entries following each other form a
-  // signature.
-  OS << "static const unsigned short SignatureTable[] = {\n";
+  // of a struct ProgModelTypeStruct table. Multiple entries following each
+  // other form a signature.
+  OS << "const unsigned short " << ClassName << "::SignatureTable[] = {\n";
```

Comment:
Why `static` was removed?

Reply:
upstream emit the lookup data as simple static data (as only OpenCL is of any interest).

`SignatureTable` (others) are now static fields of a class and this the implementation. The declaration is here https://github.com/intel/llvm/pull/1345/files/b07b2a86fa88a176c8c3115e35c0742fd4d805f3#diff-4326db80135f293472deaabc7779589fR365

Reply:
I see. Thanks for clarification.